### DISCUSSION TITLE
STL version shipped with MSVC build tools

### DISCUSSION THREAD

**Discussion by huangqinjin** [microsoft/STL#2407] [Q&A] [answered]
This repo dosen't have any tags, is there any information about which commit is shipped with different build tools?

![image](https://user-images.githubusercontent.com/8402260/145688411-783355a5-9b28-4f33-8865-187fdfda7afe.png)

**Comment by StephanTLavavej** [comment, marked as answer]
@barcharcraz notes that although MSVC and VS's sources are stored in internal git repos, their commits aren't related to commits in this repo (the histories are totally different). However, the contents of the STL's subdirectory in the MSVC repo do closely correspond to commits in this repo - we attempt to keep them binary-identical at all times with synchronized updates. Occasionally we do need to backport bugfixes to VS servicing updates "out of sequence" (whereas this repo's history is always linear), but the correspondence is generally quite close. (With the upcoming exception of a *major* backport to 16.11.x for C++20 ABI stabilization.)

We haven't created tags/releases here yet, because I still need to finalize the (ancient) 16.5 changelog which is currently incomplete. However, we are maintaining a list of which microsoft/STL commits closely correspond to the content of VS releases, see #386 for commit hashes/links.

Hope this helps!

**Reply by AshleyT3** [reply]
I've posted the question in this reply to a new discussion as https://github.com/microsoft/STL/discussions/3060. It seems related to knowledge of source matching released binaries... if you have thoughts, would be great to hear. Thanks.
